core/src/action.cpp

```diff
@@ -138,3 +138,42 @@ km_core_actions * km::core::action_item_list_to_actions_object(
 
   return actions.release();
 }
+
+
+// TODO: this is effectively the inverse of action_item_list_to_actions_object,
+//       and perhaps we should consider changing that function to be a member
+//       of state also, so that we can move memory management into state?
+bool km::core::state::set_actions(
+  km_core_actions const &actions
+) {
+  _actions.clear();
+
+  // number of codepoints (not codeunits!) to delete from app context.
+
+  for(unsigned int i = 0; i < actions.code_points_to_delete; i++) {
+    _actions.push_backspace(KM_CORE_BT_CHAR, 0); // expected value is not known
```

Comment:
so… unlike the similar function in ldml_processor, this one won't push call `_actions.push_backspace(KM_CORE_BT_MARKER, …)` if it's a marker.  Or am I getting mixed up and there are no markers in these actions anymore?

Reply:
ok no markers here.  Could note that here.

Reply:
no markers in actions -- they are only internal to the cached context now.